src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  sourceString.replace(/;\s*;/g, ' ').trim();
+
+  const styleRules = sourceString.split(';');
+
+  const result = {};
+
+  styleRules.forEach((element) => {
+    const [key, value] = element.split(':', 2);
+
+    if (value !== undefined) {
+      result[key.trim()] = value.trim();
```

Comment:
This condition allows for keys with no values to be added to the result object with `undefined` values. You should check if both key and value are not empty after trimming to avoid adding incomplete style declarations.